backend/vouch/models.py

```diff
@@ -0,0 +1,60 @@
+from django.db import models
+
+from core.models.user import User
+
+
+
+class Voucher(models.Model):
+    """A `Voucher` given by a user to another one.
+    A `Voucher` represents a mark of trust related to a specific attribute (PoP, degree, diploma, ...)
+    a user is able to grant to another one. Each mark of trust has a
+    trust value and an associated uncertainty to weight its effect.
+    These vouchers can be used by an algorithm to compute the global trust score given to every user by the
+    network, in order to weight their voting right accordingly.
+    """
+
+    by = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        help_text="The user giving the voucher.",
+        related_name="vouchers_given",
+    )
+    to = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        help_text="The user receiving the voucher.",
+        related_name="vouchers_received",
+    )
+    is_public = models.BooleanField(
+        default=False, help_text="Should the voucher be public?"
+    )
+    trust_value = models.FloatField(
+        default=0,
+        help_text="The trust value given by the vouching user to receiving user.",
+    )
+    # XXX: What is the vouching score described in the help text? Is it the
+    # `trust_value`? If yes we should use the same vocabulary to avoid any
+    # confusion.
+    uncertainty = models.FloatField(
+        default=0,
+        help_text="Uncertainty about the voucher score.",
+    )
+    attribute = models.TextField(
+        max_length=32,
+        help_text="Name of the attribute corresponding to this voucher.",
+        db_index=True,
+    )
```

Comment:
To make things more explicit about which attributes are supported, I would recommend to define a list of choices that are accepted in this field. 

For now, maybe we could even define an enum with a single choice:
```py
class VoucherAttribute(models.TextChoices):
    IDENTITY = "identity"
```
and set `choices=VoucherAttribute.choices` on this TextField.
See https://docs.djangoproject.com/en/4.0/ref/models/fields/#enumeration-types for more details.